components/infobox/wikis/valorant/infobox_character_custom.lua

```diff
@@ -0,0 +1,73 @@
+---
+-- @Liquipedia
+-- wiki=valorant
+-- page=Module:Infobox/Character/Custom
+--
+-- Please see https://github.com/Liquipedia/Lua-Modules to contribute
+--
+
+local Array = require('Module:Array')
+local Class = require('Module:Class')
+local ClassTypes = require('Module:ClassTypes')
+local Lua = require('Module:Lua')
+local Namespace = require('Module:Namespace')
+local String = require('Module:StringUtils')
+
+local Injector = Lua.import('Module:Infobox/Widget/Injector')
+local Character = Lua.import('Module:Infobox/Character')
+
+local Widgets = require('Module:Infobox/Widget/All')
+local Cell = Widgets.Cell
+
+---@class OverwatchHeroInfobox: CharacterInfobox
+local CustomCharacter = Class.new(Character)
+local CustomInjector = Class.new(Injector)
+
+---@param frame Frame
+---@return Html
+function CustomCharacter.run(frame)
+	local character = CustomCharacter(frame)
+	character:setWidgetInjector(CustomInjector(character))
+	character.args.informationType = 'Agent'
+	return character:createInfobox()
+end
+
+---@param id string
+---@param widgets Widget[]
+---@return Widget[]
+function CustomInjector:parse(id, widgets)
+	local args = self.caller.args
+	if id == 'class' then
+		Cell{name = 'Class', content = {CustomCharacter._getClassType(args)}}
```

Comment:
either use table.insert here or return

```lua
	if id == 'class' then
		table.insert(widgets, Cell{name = 'Class', content = {CustomCharacter._getClassType(args)}})
```

```lua
	if id == 'class' then
		return {Cell{name = 'Class', content = {CustomCharacter._getClassType(args)}}}
```

Reply:
While this fixes the cells from being duplicated in the infobox, it removed the icon from being displayed before the text

Reply:
can you provide a test page, a screenshot of how it shzould lok and how it looks atm

Reply:
How it looks now 
![image](https://github.com/Liquipedia/Lua-Modules/assets/58013431/35caf23b-dc84-4360-9e3b-d6216fd81345)
How it should look

Test page: https://liquipedia.net/valorant/User:Fenrir.SPAZ/test

Reply:
i suggest to decide on the input param and stick with it ;)
https://liquipedia.net/valorant/index.php?title=Module%3ASandbox%2FFenrir&type=revision&diff=896793&oldid=895545

Reply:
*SIGH*

on the bright side, its always something stupidly simple with me :D

Would you suggest having light/dark mode icons for this, or letting CSS handle the conversion?

Reply:
are they monotone?

Reply:
Yes. Currently they are all uploaded as white icons https://liquipedia.net/valorant/Module:ClassTypes/Data

Reply:
in that case add the css adjust to this pr